backend/src/main/java/com/huellapositiva/domain/actions/CancelProposalAction.java

```diff
@@ -0,0 +1,32 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.domain.model.valueobjects.ProposalStatus;
+import com.huellapositiva.domain.repository.ProposalRepository;
+import com.huellapositiva.infrastructure.orm.entities.JpaProposalStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+
+@Service
+public class CancelProposalAction {
+
+    @Autowired
+    ProposalRepository proposalRepository;
+
+    /**
+     * This method changes the specified proposal status to CANCELLED.
+     *
+     * @param id Proposal id
+     *
+     * (method "updateProposalStatus" returns 0 in case it doesn't find any proposal with that id)
+     */
+    public void executeByReviser(String id) {
+        JpaProposalStatus jpaProposalStatus = JpaProposalStatus.builder()
+                .id(ProposalStatus.CANCELLED.getId())
```

Comment:
I wouldn't rely on the hardcoded id's from an enum, they could be misaligned with the entries on the DB. You should use the JpaProposalStatusRepository to retrieve the correct ID.

Additionally, since the IDs of the statuses are static (they do not change) we could even cache the response from the DB to avoid making a DB query everytime we need to retrieve the ids of the statuses

Reply:
I suggested the change so I am guilty for this one :-)
IMHO we shouldn't let the database (infrastructure) drive model identifiers but the other way around. Because this class is in the Model, it should talk "model language" and it's identifiers.
If the values differ in the database, the adapter subscribed to that port should be the one performing the transformation.
What am I not understanding here???

Reply:
Hello guys, 
in this case, the hardcoded enum will be always updated with the DB because that table is every time created from the enum, and not modified at all during the execution, so there won't be any problem with misaligned information.
So far, in the code we are not accessing the DB every time we need to check the id of a specific proposalStatus,  if we do that, then, we should think of what you guys are saying, and decide if we want to query the DB every time or we want to cache the response and access to it when that information is needed.

What do you think is better in this case? Should we cache the proposalStatus table from the DB and update the code to access to it every time needed or leave it like that and trust the hardcoded enum (which unless there's any change, will always match with the info from the DB)?

Reply:
@manumoranchel I agree with you. The thing is that the ID of the ProposalStatus domain model could already be the enum String value itself and don't require any additional integers at all in your domain model. An integer is something meaningful for a database more than a human in many cases, so sometimes it might not add much value to the domain semantics if used as a hardcoded ID.

Reply:
oh I see what you mean. Thanks for the explanation :-)
I was talking to Maria about the change via chat. How about we merge the PR with the changes and we add a technical debt User Story to refactor this functionality keeping the Domain objects numberless and adapting when accessing the database?

Reply:
Sure.